pkg/envconf/config.go

```diff
@@ -130,16 +131,11 @@ func (c *Config) NewClient() (klient.Client, error) {
 // are confident in the configuration or call NewClient() to ensure its
 // safe creation.
 func (c *Config) Client() klient.Client {
-	if c.client != nil {
-		return c.client
-	}
-
-	client, err := klient.NewWithKubeConfigFile(c.kubeconfig)
+	client, err := c.NewClient()
 	if err != nil {
-		panic(fmt.Errorf("envconfig: client failed: %w", err).Error())
+		panic(err)
 	}
-	c.client = client
-	return c.client
+	return client
 }
```

Comment:
⚠️ I actually realised this would prevent injecting a client without a kubeconfig in the Config. As we would always have to go through `klient.NewWithKubeConfigFile(c.kubeconfig)` instead of relying on the injected one, which afaict is a supported usecase at the moment!

Reply:
See https://github.com/kubernetes-sigs/e2e-framework/pull/413 for an attempt at solving this

Reply:
Do you have a real example of this being an issue? 
Because currently, I can't figure out a situation where this happens. And you can still do `GetClient` to retrieve the client right?

Reply:
It's just that this would be a behavior change, currently someone creating a config with an injected client would always get that same client when calling `Client()`. With this change one would have to call `GetClient()` explicitly to get that back, while calling `Client()` would result in an error if there was no kubeconfig configured for example, which is something totally acceptable if you just injected your own client.

Reply:
ok, it makes sense then. If you can find a way to support this without overcomplicating thing, I'm all in. 
Otherwise, maybe we could consider doing a breaking change? Since we are running in version `v0.x.y`, I feel like we can afford to do breaking changes if it means we have more stable foundations to run the e2e tests at scale.

Reply:
See my draft PR, unfortunately I couldn't get the tests to run yet.